src/cls/rbd/cls_rbd_types.h

```diff
@@ -73,13 +74,81 @@ struct MirrorImage {
   static void generate_test_instances(std::list<MirrorImage*> &o);
 
   bool operator==(const MirrorImage &rhs) const;
+  bool operator<(const MirrorImage &rhs) const;
 };
 
 std::ostream& operator<<(std::ostream& os, const MirrorImageState& mirror_state);
 std::ostream& operator<<(std::ostream& os, const MirrorImage& mirror_image);
 
 WRITE_CLASS_ENCODER(MirrorImage);
 
+enum MirrorImageStatusState {
+  MIRROR_IMAGE_STATUS_STATE_UNKNOWN         = 0,
+  MIRROR_IMAGE_STATUS_STATE_ERROR           = 1,
+  MIRROR_IMAGE_STATUS_STATE_SYNCING         = 2,
+  MIRROR_IMAGE_STATUS_STATE_STARTING_REPLAY = 3,
+  MIRROR_IMAGE_STATUS_STATE_REPLAYING       = 4,
+  MIRROR_IMAGE_STATUS_STATE_STOPPING_REPLAY = 5,
+  MIRROR_IMAGE_STATUS_STATE_STOPPED         = 6,
+};
+
+enum MirrorImageStatusFlag {
+  MIRROR_IMAGE_STATUS_FLAG_UP               = 1<<0,
+  MIRROR_IMAGE_STATUS_FLAG_DOWN             = 1<<1,
```

Comment:
Actually, it can be up, down or unknown. That is why I decided to have two flags. Not sure how useful it is, it was expected to be used only internally, but right now does not look very useful. In 'RBD API', I use bool:
```
  typedef struct {
    mirror_image_status_state_t state;
    std::string description;
    time_t last_update;
    bool up;
  } mirror_image_status_t;
```
So may be use bool instead of flags for internal structure too?